server/datastore/mysql/activities.go

```diff
@@ -106,6 +107,40 @@ func (ds *Datastore) ListActivities(ctx context.Context, opt fleet.ListActivitie
 		return nil, nil, ctxerr.Wrap(ctx, err, "select activities")
 	}
 
+	// Fetch details as a separate query due to sort buffer issue triggered by large JSON details entries. Issue last reproduced on MySQL 8.0.36
+	// https://stackoverflow.com/questions/29575835/error-1038-out-of-sort-memory-consider-increasing-sort-buffer-size/67266529
+	IDs := make([]uint, 0, len(activities))
+	for _, a := range activities {
+		IDs = append(IDs, a.ID)
+	}
+	detailsStmt, detailsArgs, err := sqlx.In("SELECT id, details FROM activities WHERE id IN (?)", IDs)
+	if err != nil {
+		return nil, nil, ctxerr.Wrap(ctx, err, "Error binding activity IDs")
+	}
+	type activityDetails struct {
+		ID      uint             `db:"id"`
+		Details *json.RawMessage `db:"details"`
+	}
+	var details []activityDetails
+	err = sqlx.SelectContext(ctx, ds.reader(ctx), &details, detailsStmt, detailsArgs...)
+	if errors.Is(err, sql.ErrNoRows) {
```

Comment:
`sqlx.Select...` will not return `ErrNoRows`, only `sqlx.Get` does (because it expects a single row and fails if it doesn't exist).